### ISSUE TITLE
Minimum Version 15.3.3 Doesn't Work

### ISSUE THREAD

**Issue by YipYipX4** [Microsoft/vswhere#99]
I have a PowerShell script that installs various tools to ensure all developers have the same set of tools.  The script mostly uses Chocolatey and NPM, but for VS2017 I use vswhere.exe to check for a minimum version and prompt to install or update if needed.  I tried setting the minimum version to 15.3.3 when I had version 15.3.2 installed and it still found my installation (and therefore, my script did not prompt to install the update).  If I set the minimum required version to "15.4" then it finds no such version installed and my script prompts to run the installer (but then it will keep doing so because the latest available right now is 15.3.3).

Here's the relevant portion of my script:
```powershell
$here = if ($PSScriptRoot) { $PSScriptRoot } else { "." }
$minRequiredVS2017Version = "15.3.3"
$vs2017path = &"$here\vswhere.exe" -latest -version "[$minRequiredVS2017Version,99.0)" -property installationPath

if (-not $vs2017path) {
    # Prompt to run VS2017 installer (and run it)
    exit
}
```

**Comment by heaths** [comment]
The build number of Visual Studio does not work like that. It's actually a date-based build number. The .3 is more of a branding version to indicate a servicing release.

While we can't really change this (there are far too many layers that depend on the real build version),  we could perhaps find a way to publish a table of how the versions relate

**Comment by YipYipX4** [comment]
So, will vswhere.exe work correctly with the real build version?

I checked the about dialog and it just says that I now have 15.3.3, so I don't know what to try.  I've already installed 15.3.3, but I could try the real build version and then increment that by 1 to verify that it no longer finds it.

**Comment by heaths** [comment]
That dialog gets values from a different location that pertains only to branding. If you already have 15.3.3 and indeed require a fix/change in 15.3.3 or newer, run _vswhere_ on your machine to find the `installationVersion`. For example, on one of my machines with 15.3.3 _vswhere_ shows me "15.3.26730.12", so if you run the following command you will find 15.3.3 or newer.

```
vswhere -version 15.3.26730.12
```

**Comment by YipYipX4** [comment]
Ok, thanks!  I should have realized that vswhere itself can tell me the actual version since I have already installed the "15.3.3" branded update on my own system.  It would be nice to have a table of VS versions in the readme.md for this repo for reference, but I have what I need for now, so I will close this issue.

Using a minimum required version of "15.3.26730.12" in my script finds my install and if I change it to "15.3.26730.13", then VS is not found and my script prompts that VS is out-of-date and it will run the installer.
